### PR TITLE
Contrastive learning

### PR DESCRIPTION
**Pull request by guillaumekln**

### CODE REVIEW

**Comment by ChawDoe** [comment]
Hi, 
could you tell me the settings of this contrastive learning method?
Such as batch size, learning rate and the other parameters. 
I failed to reproduce the result in the same dataset mentioned in this paper. 
Thanks.

**Comment by guillaumekln** [comment]
Hi,

You can run the contrastive fine tuning with the same parameters that you used for the initial training. We reproduced some gains after 300 steps of contrastive learning with η = 0.1 and then setting the length penatly to 2.5.

Note that you should also configure the following parameters depending on your tokenization so that the noise is applied at the word-level and not the subword-level:

```yaml
params:
  # (optional) Define the subword marker. This is useful to apply noise at the word level 
  # instead of the subword level (default: ￭).
  decoding_subword_token: ￭

  # (optional) Whether decoding_subword_token is used as a spacer (as in SentencePiece) or a joiner (as in BPE).
  # If unspecified, will infer  directly from decoding_subword_token.
  decoding_subword_token_is_spacer: false
```
